functions-and-operators/control-flow-functions.md

```diff
@@ -6,11 +6,115 @@ aliases: ['/docs/dev/functions-and-operators/control-flow-functions/','/docs/dev
 | [`IF()`](https://dev.mysql.com/doc/refman/8.0/en/flow-control-functions.html#function_if)         | If/else construct                 |
 | [`IFNULL()`](https://dev.mysql.com/doc/refman/8.0/en/flow-control-functions.html#function_ifnull) | Null if/else construct            |
 | [`NULLIF()`](https://dev.mysql.com/doc/refman/8.0/en/flow-control-functions.html#function_nullif) | Return NULL if expr1 = expr2      |
+
+
+## CASE
+
+The `CASE` operator allows you to respond on multiple values and conditions.
+
+```sql
+WITH RECURSIVE d AS (SELECT 1 AS n UNION ALL SELECT n+1 FROM d WHERE n<10)
+SELECT n, CASE WHEN n MOD 2 THEN "odd" ELSE "even" END FROM d
+```
+
+```
++----+----------------------------------------------+
+| n  | CASE WHEN n MOD 2 THEN "odd" ELSE "even" END |
++----+----------------------------------------------+
+|  1 | odd                                          |
+|  2 | even                                         |
+|  3 | odd                                          |
+|  4 | even                                         |
+|  5 | odd                                          |
+|  6 | even                                         |
+|  7 | odd                                          |
+|  8 | even                                         |
+|  9 | odd                                          |
+| 10 | even                                         |
++----+----------------------------------------------+
+10 rows in set (0.00 sec)
+```
+
+## IF()
+
+The `IF()` statement allows you to do something based on wheter a value or expression is true or not.
+
+```sql
+WITH RECURSIVE d AS (SELECT 1 AS n UNION ALL SELECT n+1 FROM d WHERE n<10)
+SELECT n, IF(n MOD 2, "odd", "even") FROM d;
+```
+
+```
++----+----------------------------+
+| n  | IF(n MOD 2, "odd", "even") |
++----+----------------------------+
+|  1 | odd                        |
+|  2 | even                       |
+|  3 | odd                        |
+|  4 | even                       |
+|  5 | odd                        |
+|  6 | even                       |
+|  7 | odd                        |
+|  8 | even                       |
+|  9 | odd                        |
+| 10 | even                       |
++----+----------------------------+
+10 rows in set (0.00 sec)
+```
+
+## IFNULL()
+
+The `IFNULL()` functions returns the data of the expression if it is not NULL and if it is NULL it returns the data
+of the second argument of the function.
+
+```sql
+WITH data AS (SELECT NULL AS x UNION ALL SELECT 1 )
+SELECT x, IFNULL(x,'x has no value') FROM data;
+```
+
+```
++------+----------------------------+
+| x    | IFNULL(x,'x has no value') |
++------+----------------------------+
+| NULL | x has no value             |
+|    1 | 1                          |
++------+----------------------------+
+2 rows in set (0.0006 sec)
+```
+
+## NULLIF()
+
+The `NULLIF()` function returns null if both arguments are the same. Otherwise it returns the first argument.
```

Comment:
Can we clarify what happens when one or more of the arguments are NULL?

Reply:
```
mysql> SELECT NULLIF(1, 2);
+--------------+
| NULLIF(1, 2) |
+--------------+
|            1 |
+--------------+
1 row in set (0.00 sec)

mysql> SELECT NULLIF(1, 1);
+--------------+
| NULLIF(1, 1) |
+--------------+
|         NULL |
+--------------+
1 row in set (0.00 sec)

mysql> SELECT NULLIF(1, NULL);
+-----------------+
| NULLIF(1, NULL) |
+-----------------+
|               1 |
+-----------------+
1 row in set (0.00 sec)

mysql> SELECT NULLIF(NULL, NULL);
+--------------------+
| NULLIF(NULL, NULL) |
+--------------------+
|               NULL |
+--------------------+
1 row in set (0.00 sec)

mysql> SELECT NULLIF(NULL, 1);
+-----------------+
| NULLIF(NULL, 1) |
+-----------------+
|            NULL |
+-----------------+
1 row in set (0.00 sec)
```